rusty_pizza_server/src/order_model/meal.rs

```diff
@@ -42,6 +101,132 @@ impl<'a> Iterator for SpecialsMut<'a> {
+    meal_id: Option<String>,
+    /// Size of the pizza or noodle type etc.
+    variety: Option<String>,
+    price: Option<Money>,
+    specials: HashMap<u32, Special>,
+    special_factory: SpecialFactory,
+}
+
+impl MealBuilder {
+    pub fn new() -> MealBuilder {
+        MealBuilder {
+            meal_id: None,
+            variety: None,
+            price: None,
+            specials: HashMap::new(),
+            special_factory: SpecialFactory::new(),
+        }
+    }
+
+    /// Set meal_id of new Meal
+    pub fn meal_id<'a>(&'a mut self, meal_id: String) -> &'a mut MealBuilder {
+        self.meal_id = Some(meal_id);
+        self
+    }
+
+    /// Set variety of new Meal
+    pub fn variety<'a>(&'a mut self, variety: String) -> &'a mut MealBuilder {
+        self.variety = Some(variety);
+        self
+    }
+
+    /// Add a special to new Meal
+    pub fn special<'a>(&'a mut self, description: String) -> &'a mut MealBuilder {
+        let special = self.special_factory.create_special(description);
+        let id = special.get_id();
+        self.specials.insert(id, special);
+        self
+    }
+
+    /// Add multiple specials to new Meal
+    pub fn specials<'a>(&'a mut self, descriptions: &[String]) -> &'a mut MealBuilder {
+        let specials = descriptions
+            .iter()
+            .map(|description| self.special_factory.create_special(description.to_string()))
+            .map(|special| (special.get_id(), special))
+            .collect::<HashMap<u32, Special>>();
+        self.specials.extend(specials);
+        self
+    }
+
+    /// Set total price of new Meal
+    pub fn price<'a>(&'a mut self, price: Money) -> &'a mut MealBuilder {
+        self.price = Some(price);
+        self
+    }
+
+    /// Add a new Price to the total price to set the new total price of new Meal
+    pub fn add_price<'a>(&'a mut self, price: Money) -> &'a mut MealBuilder {
+        self.price = match self.price {
+            Some(old) => Some(old + price),
+            None => Some(price),
+        };
+        self
+    }
+
+    /// Subtract a new Price from the total price to set the new total price of new Meal
+    pub fn diff_price<'a>(
+        &'a mut self,
+        price: Money,
+    ) -> Result<&'a mut MealBuilder, MealBuilderError> {
+        self.price = match self.price {
+            Some(old) if old >= price => Some(old - price),
+            Some(old) if old < price => {
+                return Err(MealBuilderError::NegativePriceBuilded(price - old))
+            }
+            None => return Err(MealBuilderError::NegativePriceBuilded(price)),
+            _ => panic!("This should not be possible to reach"),
+        };
+        Ok(self)
+    }
+
+    /// Add a special and its price to new Meal
+    pub fn special_with_price<'a>(
+        &'a mut self,
+        description: String,
+        price: Money,
+    ) -> &'a mut MealBuilder {
+        self.special(description).add_price(price)
```

Comment:
Ah, I see why you added `add_price`. This might be a flaw in our design: Specials are not intended to be priced separately, aren't they? Do we want them to be?